chainercv/links/model/pspnet/pspnet.py

```diff
@@ -0,0 +1,476 @@
+from __future__ import division
+
+from math import ceil
+import warnings
+
+import chainer
+import chainer.functions as F
+import chainer.links as L
+from chainercv.utils import download_model
+import numpy as np
+import six
+
+try:
+    from chainermn.links import MultiNodeBatchNormalization
+except Exception:
+    warnings.warn('To perform batch normalization with multiple GPUs or '
+                  'multiple nodes, MultiNodeBatchNormalization link is '
+                  'needed. Please install ChainerMN: '
+                  'pip install pip install git+git://github.com/chainer/'
+                  'chainermn.git@distributed-batch-normalization')
+
+
+class ConvBNReLU(chainer.Chain):
+
+    def __init__(self, in_ch, out_ch, ksize, stride=1, pad=1, dilation=1):
+        super(ConvBNReLU, self).__init__()
+        comm = chainer.config.comm
+        w = chainer.config.initialW
+        with self.init_scope():
+            if dilation > 1:
+                self.conv = L.DilatedConvolution2D(
+                    in_ch, out_ch, ksize, stride, pad, dilation, True, w)
+            else:
+                self.conv = L.Convolution2D(
+                    in_ch, out_ch, ksize, stride, pad, True, w)
+            if comm is not None:
+                self.bn = MultiNodeBatchNormalization(
+                    out_ch, comm, eps=1e-5, decay=0.95)
+            else:
+                self.bn = L.BatchNormalization(out_ch, eps=1e-5, decay=0.95)
+
+    def __call__(self, x, relu=True):
+        h = self.bn(self.conv(x))
+        return h if not relu else F.relu(h)
+
+
+class PyramidPoolingModule(chainer.ChainList):
+
+    def __init__(self, in_ch, feat_size, pyramids):
+        super(PyramidPoolingModule, self).__init__(
+            ConvBNReLU(in_ch, in_ch // len(pyramids), 1, 1, 0),
+            ConvBNReLU(in_ch, in_ch // len(pyramids), 1, 1, 0),
+            ConvBNReLU(in_ch, in_ch // len(pyramids), 1, 1, 0),
+            ConvBNReLU(in_ch, in_ch // len(pyramids), 1, 1, 0))
+        if isinstance(feat_size, int):
+            self.ksizes = (feat_size // np.array(pyramids)).tolist()
+        elif isinstance(feat_size, (list, tuple)) and len(feat_size) == 2:
+            kh = (feat_size[0] // np.array(pyramids)).tolist()
+            kw = (feat_size[1] // np.array(pyramids)).tolist()
+            self.ksizes = list(zip(kh, kw))
+
+    def __call__(self, x):
+        ys = [x]
+        h, w = x.shape[2:]
+        for f, ksize in zip(self, self.ksizes):
+            y = F.average_pooling_2d(x, ksize, ksize)  # Pad should be 0!
+            y = f(y)  # Reduce num of channels
+            y = F.resize_images(y, (h, w))
+            ys.append(y)
+        return F.concat(ys, axis=1)
+
+
+class BottleneckConv(chainer.Chain):
+
+    def __init__(self, in_ch, mid_ch, out_ch, stride=2, dilate=False):
+        mid_stride = chainer.config.mid_stride
+        super(BottleneckConv, self).__init__()
+        with self.init_scope():
+            self.cbr1 = ConvBNReLU(
```

Comment:
I want to make the attribute name of `ConvBN*` consistent across the library.
We have `Conv2DBNActiv`, which will be used by ResNet implementations.

Since `cbr` is specific to `Conv->BN->ReLU`, I want to avoid the name.
How about `conv1`, `conv2`, ... ?

Reply:
Thanks, I changed the attribute name to `conv1`, `conv2`, ...